teuthology/run_tasks.py

```diff
@@ -71,27 +158,70 @@ def run_one_task(taskname, **kwargs):
 
 def run_tasks(tasks, ctx):
     archive_path = ctx.config.get('archive_path')
+    tasks_dir = "tasks"
+    log_path = str()
+    log_level = str()
+    log_formatter = str()
+
     if archive_path:
+        log_path, log_formatter, log_level = \
+            configure_logging(ctx, archive_path, tasks_dir)
         timer = Timer(
             path=os.path.join(archive_path, 'timing.yaml'),
             sync=True,
         )
     else:
         timer = Timer()
     stack = []
+    all_tasks = []
     try:
         for taskdict in tasks:
             try:
                 ((taskname, config),) = taskdict.items()
             except (ValueError, AttributeError):
                 raise RuntimeError('Invalid task definition: %s' % taskdict)
+
```

Comment:
I have feeling that there is too much support code added, which hides base idea of cycle. Probably, function should be created and log support code should be moved in it.